payjoin/src/receive/v2/mod.rs

```diff
@@ -568,4 +568,31 @@ mod test {
         let deserialized: Receiver = serde_json::from_str(&serialized).unwrap();
         assert_eq!(session, deserialized);
     }
+
+    #[test]
+    fn test_v2_pj_uri() {
+        let address = bitcoin::Address::from_str("12c6DSiU4Rq3P4ZxziKxzrL5LmMBrzjrJX")
+            .unwrap()
+            .assume_checked();
+        let receiver_keys = crate::hpke::HpkeKeyPair::gen_keypair();
+        let ohttp_keys =
+            OhttpKeys::from_str("OH1QYPM5JXYNS754Y4R45QWE336QFX6ZR8DQGVQCULVZTV20TFVEYDMFQC")
+                .expect("Invalid OhttpKeys");
+        let arbitrary_url = Url::parse("https://example.com").unwrap();
+        let uri = Receiver {
+            context: SessionContext {
+                address,
+                directory: arbitrary_url.clone(),
+                subdirectory: None,
+                ohttp_keys,
+                ohttp_relay: arbitrary_url.clone(),
+                expiry: SystemTime::now() + Duration::from_secs(60),
+                s: receiver_keys,
+                e: None,
+            },
+        }
+        .pj_uri();
+        assert_ne!(uri.extras.endpoint, arbitrary_url);
+        assert!(!uri.extras.disable_output_substitution);
+    }
```

Comment:
based on [coveralls's comment](https://github.com/payjoin/rust-payjoin/pull/466#issuecomment-2574064297) this doesn't seem to offer us additional coverage beyond the integration test. Is it the right approach to testing @spacebear21 ?